DEVELOPMENT.md

```diff
@@ -10,6 +10,104 @@ This document outlines the process for setting up the development environment fo
 
 ## Setting up the Environment
 
+Decide first if you'd like a full buildsystem (on a VM) or a containerized dev environment.
+
+### VM as buildsystem
+Uses a Ubuntu 24.04 as base to run `chef` to setup all dependencies.
+The initial compilation is CPU intense and `16vcpu` are recommended.
+This was tested on GCP: a balanced disk of 500 GB and a VM type that supports nested virtualization should be chosen, as of writing (May 2025)
+`n2-standard-16` works well. 
+
+> [!Warning]
+>  The first build takes several hours and at least 160 Gb of space
+
+Turn on nested virtualization and avoid the use of `spot` VMs for the first build to avoid the very long first
+build interrupting. If you create the VMs as templates from an image, you can later switch to more cost-effective `spot` instances.
+
+
+
+```yaml
+advancedMachineFeatures:
+  enableNestedVirtualization: true
+```
+
+1) Install chef and some dependencies
+ 
+```
+sudo apt update sudo apt install -y git coreutils mkcert libnss3-tools screen libvirt-daemon-system libvirt-clients qemu-kvm virt-manager
+curl -L https://chefdownload-community.chef.io/install.sh | sudo bash
+```
+Now, on this VM, clone pixie (or your fork of it)
+
+```
+git clone https://github.com/pixie-io/pixie.git
+cd pixie/tools/chef
+sudo chef-solo -c solo.rb -j node_workstation.json
+sudo usermod -aG libvirt $USER
+```
+
+Make permanent the env loading via your bashrc
+```sh
+echo "source /opt/px_dev/pxenv.inc " >> ~/.bashrc
+```
+
+Put the baselrc into your homedir:
+```sh
+cp .bazelrc ~/.
+```
+In order to very significantly speed up your work, you may opt for a local cache directory. This can be shared between users of the VM, if both are part of the same group.
+Create a cache dir under <directory-path> like /tmp/bazel
+```sh
+sudo groupadd bazelcache
+sudo usermod -aG bazelcache $USER
+sudo mkdir -p <directory-path>
+sudo chown :bazelcache <directory-path>
+sudo chmod 2775 <directory-path>
+```
+
+2) Create/Use a registry you control and login
+   
+```sh
+docker login ghcr.io/<myregistry>
+```
+
+3) Make Minikube run and deploy a vanilla pixie
```

Comment:
```suggestion
3) Prepare your Kubernetes cluster and deploy an existing Pixie release
```

Could we adjust the wording to emphasize that development can be done against any Kubernetes cluster? That way, users with an existing cluster won’t feel the need to set up an additional one.

Maybe something like this:

> If you have an existing kubernetes cluster to develop against, ensure your kubectl config is set properly. Otherwise, you can create a minikube cluster .... (existing instructions here).

Reply:
Very good point, yes. will add. 

From a maintainer PoV: wouldnt you want ppl to use a golden path, though? To limit the range of potential issues?

Reply:
I added the destinction in two WARNING boxes